tests/functional/interfaces-with-arguments.ts

```diff
@@ -0,0 +1,105 @@
+import "reflect-metadata";
+import { getMetadataStorage } from "../../src/metadata/getMetadataStorage";
+import {
+  IntrospectionSchema,
+  IntrospectionObjectType,
+  IntrospectionInterfaceType,
+  IntrospectionField,
+} from "graphql";
+import { getSchemaInfo } from "../helpers/getSchemaInfo";
+import { ArgsType, Field, Int, InterfaceType, ID, Query, ObjectType, Arg, Args } from "../../src";
+
+describe("Interfaces with arguments", () => {
+  describe("Schema", () => {
+    let schemaIntrospection: IntrospectionSchema;
+    let queryType: IntrospectionObjectType;
+
+    beforeAll(async () => {
+      getMetadataStorage().clear();
+
+      @ArgsType()
+      class ImageArgs {
+        @Field(type => Int, { nullable: true })
+        width?: number;
+        @Field(type => Int, { nullable: true })
+        height?: number;
+      }
+
+      @InterfaceType()
+      abstract class Article {
+        @Field(type => ID)
+        id: string;
+        @Field({ nullable: true })
+        headline?: string;
+
+        @Field()
+        image1(@Arg("width") width: number, @Arg("height") height: number): string {
+          return `http://lorempixel.com/${width}/${height}/`;
+        }
+        @Field()
+        image2(@Args(type => ImageArgs) args: any): string {
```

Comment:
I think one sample field method is enough for testing purposes

Reply:
I added multiple to test the difference between inline arguments and using an arguments type. And it even turned out to be quite useful in the newly added functional tests that were suggested by you.

Reply:
Ok, I haven't spotted that because of not meaningful field names 😉